### PR TITLE
drivers/lwnl: fix lwnl return type

### PR DESCRIPTION
**Pull request by pillip8282**
fix lwnl return type

### CODE REVIEW

File: os/fs/vfs/fs_select.c [hunk 1]
```diff
@@ -266,7 +266,7 @@ int select(int nfds, FAR fd_set *readfds, FAR fd_set *writefds, FAR fd_set *exce
 
 	DEBUGASSERT(ndx == npfds);
 	if (ndx != npfds) {
-		set_errno(EINVAL);
+		set_errno(EBADF);
```
Comment (jeongarmy) [inline os/fs/vfs/fs_select.c:269]: How about keeping EINVAL? In posix, a select set errno as EINVAL when the nfds argument is less than 0 or greater than FD_SETSIZE.

Comment (pillip8282) [reply os/fs/vfs/fs_select.c:269]: ndx != npfds doesn't mean that nfds is negative or  exceeds the RLIMIT_NOFILE resource limit (see getrlimit(2)). Hence we don't need to modify it

File: os/drivers/lwnl/lwnl.c [hunk 2]
```diff
@@ -122,6 +121,11 @@ static int lwnl_open(struct file *filep)
 	struct lwnl_upperhalf_s *upper = inode->i_private;
 
 	LWNLDEV_LOCK(upper);
+	/*
+	 * crefs type is uint8 so overflow might happen.
+	 * However there are a few modules which use lwnl.
+	 * So checking overflow is not needed now.
```
Comment (davidfather) [inline os/drivers/lwnl/lwnl.c:127]: Do you mean that overflow checking will be done later by some modules which use lwnl?

File: os/drivers/lwnl/lwnl.c [hunk 3]
```diff
@@ -204,7 +212,7 @@ static int lwnl_ioctl(struct file *filep, int cmd, unsigned long arg)
 	LWNL_ENTER;
 	int res = lwnl_add_listener(filep);
 	if (res < 0) {
-		res = -ENOSYS;
+		res = -EBADF;
```
Comment (davidfather) [inline os/drivers/lwnl/lwnl.c:215]: In this function, there are two cases where the return type is set to '-EBADF'. How could you tell them apart?

File: os/drivers/lwnl/lwnl.c (hunk 2 above)
Comment (pillip8282) [reply os/drivers/lwnl/lwnl.c:127]: I'll modify it if many modules use lwnl. However only wi-fi manager and iotivity access lwnl now. So I think we don't need to add overflow logic now.

File: os/drivers/lwnl/lwnl.c (hunk 3 above)
Comment (pillip8282) [reply os/drivers/lwnl/lwnl.c:215]: According to ioctl manual page. ioctl can return following errno values.
---
EBADF  fd is not a valid file descriptor.

       EFAULT argp references an inaccessible memory area.

       EINVAL request or argp is not valid.

       ENOTTY fd is not associated with a character special device.

       ENOTTY The specified request does not apply to the kind of object that the file descriptor fd references
---
There is no relevant errno value for this. If you have good idea, I'll apply it.

**Review by davidfather — APPROVED** [review]
Looks good to me.

**Pull request merged**